src/modules/rlm_lua/lua.c

```diff
@@ -466,6 +479,28 @@ static int _lua_list_iterator_init(lua_State *L)
 	return 1;
 }
 
+static int _lua_fr_rcode_table_index(lua_State *L)
+{
+	FR_NAME_NUMBER const *this = fr_rcode_table;
+	char const *key = lua_tostring(L, -1);
+
+	for (; this->name != NULL; this++)
+		if (!strcmp(this->name, key)) {
+			lua_pushstring(L, this->name);
+			break;
+		}
+
+	return 1;
+}
+
+static void rlm_lua_register_rcode_table(lua_State *L, char const *name)
```

Comment:
Just create a table, don't have it call a c function every time it wants to return a value, that's horribly slow. If you want to make the table read-only use a proxy table. http://lua-users.org/wiki/ReadOnlyTables

Reply:
the pr was updated.